### PR TITLE
ENH add private arg to push

### LINKED ISSUE #112: Add `private: bool` to hub_utils.push
Probably a good idea to allow users to push to a private repo if they want to, by accepting a `private: bool` attribute on `hub_utils.push`.

### PR DESCRIPTION
**Pull request by adrinjalali**
Fixes #112 and #114

### CODE REVIEW

**Comment by adrinjalali** [comment]
@narsil @osanseviero do we not want to support private repos on the inference side? I wanted to have all repos created by the CI to be private, but we can't test the inference engine if we make them private.

**Comment by adrinjalali** [comment]
I'd say we can merge (if the CI is green) and make the inference test repo private once/if we support that in the backend side.

cc @skops-dev/maintainers

File: skops/hub_utils/_hf_hub.py [hunk 1]
```diff
@@ -561,7 +571,10 @@ def download(
     dst = Path(dst)
     if dst.exists() and bool(next(dst.iterdir(), None)):
         raise OSError("None-empty dst path already exists!")
-    dst.rmdir()
+
+    # remove the folder only if it's empty and it exists
+    if dst.exists():
```
Comment (BenjaminBossan) [inline skops/hub_utils/_hf_hub.py:576]: I'm a bit astonished that the missing check did not cause trouble so far.

**Review by BenjaminBossan — APPROVED** [review]
Looks good to me, only found a typo.

File: skops/hub_utils/_hf_hub.py (hunk 1 above)
Comment (adrinjalali) [reply skops/hub_utils/_hf_hub.py:576]: same

**Comment by adrinjalali** [comment]
adding some more docs ...

**Comment by adrinjalali** [comment]
Now done!

**Pull request merged**